packages/configraph/src/pick.ts

```diff
@@ -0,0 +1,177 @@
+import * as _ from 'lodash-es';
+
+import { ConfigraphDataType } from './data-types';
+import { createResolver, DependencyNotResolvedResolutionError } from './resolvers';
+import { ConfigraphNode } from './config-node';
+import { SchemaError } from './errors';
+import { templateContextAls } from './entity-template';
+
+// special new data type that will delegate calls back to the original "picked" data type
+export class PickedDataType extends ConfigraphDataType {
```

Comment:
pick is a little different than something like inject - which is just a resolver. Pick needs to set all of the data type properties, not just how it is resolved, so we need to be able to link to the source node.

While the line here between data types / config nodes / resolvers all gets a bit blurrier than I would like, I think the end result for users is fairly clean and intuitive.